climada/engine/forecast.py

```diff
@@ -0,0 +1,897 @@
+        else:
+            self.exposure_name = exposure_name
+        self.vulnerability = impact_funcs
+        self._impact = [Impact() for i in range(len(self.run_datetime))]
+
+    def ei_exp(self, run_datetime=None):
+        """
+        Expected annual impact
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        float
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return self._impact[haz_ind].eai_exp
+
+    def ai_agg(self, run_datetime=None):
+        """ Annual average impact
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        float
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return self._impact[haz_ind].aai_agg
+
+    def haz_summary_str(self, run_datetime=None):
+        """ provide a summary string for the hazard part of the forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        str
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return (self.hazard[haz_ind].tag.haz_type +
+                '_' +
+                self.haz_model +
+                '_run' +
+                run_datetime.strftime('%Y%m%d%H') +
+                '_event' +
+                self.event_date.strftime('%Y%m%d')
+                )
+
+    def summary_str(self, run_datetime=None):
+        """ provide a summary string for the impact forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        str
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        return (self.haz_summary_str(run_datetime) +
+                '_' +
+                self.exposure_name
+                )
+
+    def lead_time(self, run_datetime=None):
+        """ provide the lead time for the impact forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            select the used hazard by the run_datetime.
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        return self.event_date - run_datetime
+
+    def calc(self, force_reassign=False, check_plot=False):
```

Comment:
Please put the `check_plot` in a separate `check` method or remove entirely. There is no need for debugging functionality in the code.